### PR TITLE
[stable/3.0] Remove target_platform and default_os valus from admin node

### PR DESCRIPTION
**Pull request by jsuchome**
We need it to be filled by new values after the OS upgrade.

`admin_node["provisioner"]["default_os"]` is used to set the new OS for nodes after the server is upgraded

### CODE REVIEW

**Comment by jsuchome** [comment]
Surprisingly it seems to work.

**Comment by jsuchome** [comment]
This would rather belong to Cloud7 part of process, to some initial code run by cloud-init. Probably together with schema migration, like proposed https://github.com/SUSE-Cloud/automation/pull/1156#discussion_r76263439

@rsalevsky what do you think?

**Comment by rsalevsky** [comment]
@jsuchome It's definitely a C7 part, but we can't do it in crowbar-init because we can't access this data.

**Comment by jsuchome** [comment]
Hm, but where should we invoke it on Cloud7 side? Here we at least know we are starting the upgrade.

**Comment by MaximilianMeister** [comment]
I think it makes most sense for now at this stage (although I agree it should be in cloud 7, but that's not really possible now) and in case the upgrade goes wrong I guess the attributes get set back to the old os value again. Also when the user restores from the backup there shouldn't be a problem with this change.

**Comment by MaximilianMeister** [comment]
for me it's a +1

@vuntz what do you think?

**Comment by vuntz** [comment]
I also believe it should be done on the Cloud 7 side of things, and tbh, I'm confused as to why we can't do it there right now. Don't we have some code that runs at the beginning of the upgrade process in the rails app?

We can merge that now if it unblocks things, but please track somewhere the fact that it should be done in another location.

Also, if we really keep it here, then we'll need the same code in master for the 7->8 upgrade...

**Comment by MaximilianMeister** [comment]
> Don't we have some code that runs at the beginning of the upgrade process in the rails app?

i guess you mean on the cloud 7 side? I am not aware of anything like that, the first crowbar API consumed will be the node repo checks (after the database setup done by the crowbar-init api)

**Comment by MaximilianMeister** [comment]
but we could also make it part of `prepare` (https://trello.com/c/xeN0LnPm/1856-5-implement-api-for-preparing-nodes-for-upgrade-upgrade-step-2) after the chef-client daemon stop. and the attributes could be rolled back any time when the user cancels the upgrade...

**Comment by vuntz** [comment]
@MaximilianMeister let me rephrase: how will we make sure we only show the upgrade UI until the upgrade is fully completed? Don't we have code to force this?

**Comment by MaximilianMeister** [comment]
> how will we make sure we only show the upgrade UI until the upgrade is fully completed? Don't we have code to force this?

just yesterday I had the same thought with santiago, i wasn't sure so I created https://trello.com/c/MZP6krgU/1915-restrict-api-access-during-upgrade

as we are now decoupling the UI from Rails, it will be slightly different i guess (in the frontend), but I am afraid I still don't understand you. where would you put that piece of code exactly?

**Comment by vuntz** [comment]
> as we are now decoupling the UI from Rails, it will be slightly different i guess (in the frontend), but I am afraid I still don't understand you. where would you put that piece of code exactly?

If we have code that enforces the upgrade mode, then when crowbar starts, we can trigger automatic actions like the one we're discussing here.

**Comment by vuntz** [comment]
And to be clear: let's not block on this discussion forever, we can merge this PR. What matters to me is that we track the fact that we need to clean this up at a later point.

**Comment by jsuchome** [comment]
@vuntz so could you add your +1? I've created https://trello.com/c/8X7elkka/82-cleanup-upgrade-related-mess to find the proper solution later...
